### ISSUE TITLE
Questions about the recommended settings in yuzu

### ISSUE THREAD

**Issue by nova1751** [HolographicWings/TOTK-Mods-collection#147]
I found the recommend setting about the `Use asynchronous shader building (Hack)` differs in these two places,I wonder whether I should check this option or not,though I didn't run into any trouble when this option is checked.
- <img width="368" alt="image" src="https://github.com/HolographicWings/TOTK-Mods-collection/assets/97817985/ca884ae6-6cbe-4593-a20e-faec253da4ca">
- <img width="788" alt="image" src="https://github.com/HolographicWings/TOTK-Mods-collection/assets/97817985/c04408b4-1cef-4f5d-8196-ae32c794122e">

Besides,I found that the FSR option also perfoms really well when the game's resolution is equal to or higher than the screen's,which makes the game graphic much more clearer while having almost no impact on the frame rate.I'guess it's because of the shapening effect of fsr.

I hope my suggestions can be helpful.Thank you!😊😊😊

**Comment by SpockBauru** [comment]
Thanks for the tip. The asynchronous shader building make the weapons look weird in the inventory, with missing parts, but makes the game stutter a bit less. 
We just can't decide the recommendation because in one hand with the option Off the stutters are tiny on new yuzu versions, and the weapon bug is completely fixed. In other hand there are no shader stutters with the option On, but the weapon bug is really annoying.
Since yuzu changed the setting windows I will probably just update the picture with the option Off, because the weapon thing really bugs me.

**Comment by nova1751** [comment]
Thanks for your answer.I also wonder whether the FSR option can play a role in optimizing the game's graphic.I found that by turning this option on the game's graphic becomes smoother and clearer,even when the game's resolution is equal to or beyond the screen's resolution.
I think it might be a good idea to take this option into consideration when optimizing the game's graphic.I hope my suggestion can be some of help.

**Comment by nova1751** [comment]
> Thanks for the tip. The asynchronous shader building make the weapons look weird in the inventory, with missing parts, but makes the game stutter a bit less. We just can't decide the recommendation because in one hand with the option Off the stutters are tiny on new yuzu versions, and the weapon bug is completely fixed. In other hand there are no shader stutters with the option On, but the weapon bug is really annoying. Since yuzu changed the setting windows I will probably just update the picture with the option Off, because the weapon thing really bugs me.

Thanks for your answer.I also wonder whether the FSR option can play a role in optimizing the game's graphic.I found that by turning this option on the game's graphic becomes smoother and clearer,even when the game's resolution is equal to or beyond the screen's resolution.
I think it might be a good idea to take this option into consideration when optimizing the game's graphic.I hope my suggestion can be some of help.

**Comment by Ryusennin** [comment]
You should definitely keep the async shaders enabled to prevent constant stuttering.

The corrupted icons can easily be fixed by simply dropping the item and picking it up again, that will regenerate a proper thumbnail. A slight inconvenience that is certainly not worth disabling the dynamic shaders.

**Comment by SpockBauru** [comment]
> You should definitely keep the async shaders enabled to prevent constant stuttering.
> 
> The corrupted icons can easily be fixed by simply dropping the item and picking it up again, that will regenerate a proper thumbnail. A slight inconvenience that is certainly not worth disabling the dynamic shaders.

This debate is as long as the game was released. Yuzu made huge improvement on shader compilation times and is not as inconvenient as before. Please try the latest mainline, see how are things on your pc and post your pc specs and Yuzu version here. On my i7 12700 after 5 minutes of gameplay the stuttering is almost imperceptible.

**Comment by hoverbike1** [comment]
> "I found the recommend setting about the `Use asynchronous shader building (Hack)` differs in these two places, I wonder whether I should check this option or not, though I didn't run into any trouble when this option is checked."

You'll find that async shader rendering can cause temporary broken textures, which are typically fixed when you flush the shader cache (e.g. by loading the game again, restarting yuzu, etc.)
you may also experience menu icons initially being rendered incorrectly.
you may also experience certain effects, such as lasers, not being rendered correctly initially.

that sort of covers it for notes regarding which issues it typically causes.

you'll find a lot of anecdotes regarding performance with it on/off, i'll provide you some of my own, further down this post.
what i'm trying to say is that, this seems to be hardware specific. asking someone whose 3080 is loaded at about 20-30%, he will tell you that shader rendering practically has no performance impact for him. well, that is largely due to how much overhead he has, or, in other words, how many resources / how much performance he has left over, that isn't needed to emulate the game, that can be used to render those shaders.

i personally have that same 3080, just like Spock, albeit mine is from that meme company "MSI", while he actually found a decent one from a brand I haven't a clue about. 
Galax! Thats the name. I remember their branding is plastered all over the 3DMark Time Spy benchmark, haha.
Spock has a 1070 as well, which allows him further insight into midrange PC performance.

Fortunately, or unfortunate, none of us have any super low-end PC's for testing.
I mean, technically i have a GT 710, but that was more for testing that PCI works on the motherboard more than anything else.

TL;DR: You should test it for yourself, performance varies with hardware combinations. We've found that most people benefit more from disabling / not enabling this option, and thus, thats the default.

> "I also wonder whether the FSR option can play a role in optimizing the game's graphic. I found that by turning this option on the game's graphic becomes smoother and clearer, even when the game's resolution is equal to or beyond the screen's resolution."

Yuzu's FSR Window Adapting Filter is best used at the same resolution as your screen, or when Yuzu is rendering below your screens resolution, in 3D games. 2D / Sprite based games may not appear as attractive using this effect.

When rendering above your screens native resolution (e.g. rendering 3840x2160p / 4K to display on a 1080p monitor) I'd much rather recommend you use Bicubic scaling, and should you go even further beyond, such as 8K on a 1080p screen, I'd recommend Gaussian scaling, as the image can appear to have pixels warping, which gaussian seems to average out to such a degree that it becomes nearly inpercievable.

Here is Yuzu's own description of the Window Adapting Filters.
![filters](https://github.com/HolographicWings/TOTK-Mods-collection/assets/57139770/eb0c3e12-cf9f-4d33-ab93-8b92b0a3269e)

**Comment by nova1751** [comment]
> > "I found the recommend setting about the `Use asynchronous shader building (Hack)` differs in these two places, I wonder whether I should check this option or not, though I didn't run into any trouble when this option is checked."
> 
> You'll find that async shader rendering can cause temporary broken textures, which are typically fixed when you flush the shader cache (e.g. by loading the game again, restarting yuzu, etc.) you may also experience menu icons initially being rendered incorrectly. you may also experience certain effects, such as lasers, not being rendered correctly initially.
> 
> that sort of covers it for notes regarding which issues it typically causes.
> 
> you'll find a lot of anecdotes regarding performance with it on/off, i'll provide you some of my own, further down this post. what i'm trying to say is that, this seems to be hardware specific. asking someone whose 3080 is loaded at about 20-30%, he will tell you that shader rendering practically has no performance impact for him. well, that is largely due to how much overhead he has, or, in other words, how many resources / how much performance he has left over, that isn't needed to emulate the game, that can be used to render those shaders.
> 
> i personally have that same 3080, just like Spock, albeit mine is from that meme company "MSI", while he actually found a decent one from a brand I haven't a clue about. Galax! Thats the name. I remember their branding is plastered all over the 3DMark Time Spy benchmark, haha. Spock has a 1070 as well, which allows him further insight into midrange PC performance.
> 
> Fortunately, or unfortunate, none of us have any super low-end PC's for testing. I mean, technically i have a GT 710, but that was more for testing that PCI works on the motherboard more than anything else.
> 
> TL;DR: You should test it for yourself, performance varies with hardware combinations. We've found that most people benefit more from disabling / not enabling this option, and thus, thats the default.
> 
> > "I also wonder whether the FSR option can play a role in optimizing the game's graphic. I found that by turning this option on the game's graphic becomes smoother and clearer, even when the game's resolution is equal to or beyond the screen's resolution."
> 
> Yuzu's FSR Window Adapting Filter is best used at the same resolution as your screen, or when Yuzu is rendering below your screens resolution, in 3D games. 2D / Sprite based games may not appear as attractive using this effect.
> 
> When rendering above your screens native resolution (e.g. rendering 3840x2160p / 4K to display on a 1080p monitor) I'd much rather recommend you use Bicubic scaling, and should you go even further beyond, such as 8K on a 1080p screen, I'd recommend Gaussian scaling, as the image can appear to have pixels warping, which gaussian seems to average out to such a degree that it becomes nearly inpercievable.
> 
> Here is Yuzu's own description of the Window Adapting Filters. ![filters](https://user-images.githubusercontent.com/57139770/257272018-eb0c3e12-cf9f-4d33-ab93-8b92b0a3269e.png)

Thanks for you reply!Now I figured it out.
